### ISSUE TITLE
Ash.Resource does not translate into Ecto.Schema with correct autogenerate

### ISSUE THREAD

**Issue by m0rt3nlund** [ash-project/ash#708]
**Describe the bug**
I am not sure if this is a bug or a feature seen from the project, but for me it is a bug.

**To Reproduce**
- Create an ecto schema where some fields have defaults and autogenerate, like etc the timestamps
```
defmodule Shared.Config.Version do
  @moduledoc false

  schema "conf_version" do
    field(:version, :integer, default: 1)

    timestamps()
  end
end
```
- Create a Ash.Resource where some attribute has defaults
```
defmodule Shared.Ash.Config.Version do
  use Ash.Resource,
    data_layer: AshPostgres.DataLayer

  attributes do
    uuid_primary_key :id

    attribute :version, :integer do
      default 1
    end

    update_timestamp :updated_at
    create_timestamp :inserted_at
  end

  postgres do
    table "conf_version"
    repo(Shared.Repo)
  end

  actions do
    defaults [:create, :read, :update, :destroy]
  end
end
```

Now try to insert each of them using Ecto Repo.insert! and only the Ecto schema is inserted with default values and the timestamps are autogenerated

One can also use MyAshResource.__schema__(:autogenerate_fields) to see the results of my question.

**Expected behavior**
Expects the Ash.Resource to generate a Ecto.Schema that has the autogenerate options set

```
Shared.Config.Version.__schema__(:autogenerate_fields) #=> [:inserted_at, :updated_at]

Shared.Ash.Config.Version.__schema__(:autogenerate_fields) #=> []
```

```
%Shared.Config.Version{} |> dbg()

%Shared.Config.Version{} #=> %Shared.Config.Version{
  __meta__: #Ecto.Schema.Metadata<:built, "conf_version">,
  id: nil,
  version: 1,
  inserted_at: nil,
  updated_at: nil
}


%Shared.Ash.Config.Version{} |> dbg()

%Shared.Ash.Config.Version{} #=> #Shared.Ash.Config.Version<
  node: #Ash.NotLoaded<:relationship>,
  __meta__: #Ecto.Schema.Metadata<:built, "conf_version">,
  id: nil,
  version: nil,
  updated_at: nil,
  inserted_at: nil,
  aggregates: %{},
  calculations: %{},
  ...
>


```

**Comment by zachdaniel** [comment]
Agreed, you should be able to reuse the resource as an ecto schema. However, the way we do timestamps is a bit different...I'm assuming ecto needs more than just what fields are generated, i.e it needs to know that `:updated_at` is only changed on update and so on. This would need more investigation, but I imagine we can accomplish this here: https://github.com/ash-project/ash/blob/main/lib/ash/resource/schema.ex

**Comment by m0rt3nlund** [comment]
Thanks for the quick response @zachdaniel !
By the way, the project is very good!
I enjoy the DSL approach and so far it has made my project alot easier to maintain :)
So thank you for your effort!

I might contribute some features as well :)

**Comment by zachdaniel** [comment]
I probably won't have time to tackle this anytime soon, but PRs/further discussion is open :)

**Comment by m0rt3nlund** [comment]
Understandable :)
Might not be that much usage since it was not brought up allready :)

i will try to make a PR :)

**Comment by spicychickensauce** [comment]
I'm running into the same problem. I'm currently trying to migrate to ash, and so I have a lot of code that does not go through ash (yet?).
This issue might be the last? big blocker for removing the ecto models completely.
@m0rt3nlund did you maybe find a good workaround for it? Or are you still carrying both models around?

**Comment by m0rt3nlund** [comment]
> I'm running into the same problem. I'm currently trying to migrate to ash, and so I have a lot of code that does not go through ash (yet?). This issue might be the last? big blocker for removing the ecto models completely. [@m0rt3nlund](https://github.com/m0rt3nlund) did you maybe find a good workaround for it? Or are you still carrying both models around?

No, I just fully committed to Ash and refactored all my schemas into `Ash.Resource`, it was the right choice!

One solution would be to create a function that does a `Ash.Resource` -> `Ecto.Schema` conversion for the field types that do not comply.
You could pipe this for all `Repo.insert` functions until you have it all figured out.

```elixir
  myresource
  |> into_ecto()
  |> Repo.insert!()
```

**Comment by m0rt3nlund** [comment]
Actually, thinking about it.. We should make an `Ash` task that can convert existing Ecto schemas into valid `Ash.Resource`
That would help people to migrate! :)

**Comment by spicychickensauce** [comment]
@m0rt3nlund ash resources are already compatible with Repo.insert, so that `into_ecto` already exists and is build-in. But it's not fully compatible, e.g. like in the issue here the converted models are missing the correct timestamp options. Another issue is that they do not support ecto preloads. But generally it already works for most situations. 

Also, there already is a handy task that can generate Ash resources based on ecto schemas. Well, not directly, but [via the postgres db schema](https://hexdocs.pm/ash_postgres/Mix.Tasks.AshPostgres.Gen.Resources.html). I used that to help with the migration, so that wasn't an obstacle for me.

**Comment by CaydenLords** [comment]
@zachdaniel I just tried to reproduce this and it seems to be working. Is there anything that needs done here, or can this be closed?

**Comment by zachdaniel** [comment]
Ah, yeah I think you are correct someone fixed it recently 🙇
